processing/src/main/java/org/apache/druid/segment/UnnestStorageAdapter.java

```diff
@@ -285,31 +311,53 @@ void add(@Nullable final Filter filter)
             }
           }
         }
+        preFilters.add(filter);
+
+      }
+    }
 
-        if (requiredColumns.contains(outputColumnName)) {
-          // Rewrite filter post-unnest if possible.
+    final FilterSplitter filterSplitter = new FilterSplitter();
+
+    // non-unnest case
+    // OR CASE
+    List<Filter> preFilterList = new ArrayList<>();
+    List<Filter> postFilterList = new ArrayList<>();
+    if (queryFilter instanceof OrFilter) {
+      for (Filter filter : ((OrFilter) queryFilter).getFilters()) {
+        if (filter.getRequiredColumns().contains(outputColumnName)) {
           final Filter newFilter = rewriteFilterOnUnnestColumnIfPossible(filter, inputColumn, inputColumnCapabilites);
           if (newFilter != null) {
-            // Add the rewritten filter pre-unnest, so we get the benefit of any indexes, and so we avoid unnesting
-            // any rows that do not match this filter at all.
-            preFilters.add(newFilter);
+            preFilterList.add(newFilter);
+            postFilterList.add(newFilter);
```

Comment:
postFilter should be the original filter, I think?

Reply:
This has been refactored with comments